src/charm.py

```diff
@@ -174,12 +174,29 @@ def _process_wheel_file(self, event):  # noqa: C901
 
                 # Push wheel file to the container and extract it.
                 container.push(wheel_file, wheel_data, make_dirs=True)
-                container.exec(["apt-get", "update"]).wait()
-                container.exec(["apt-get", "install", "unzip"]).wait()
-                container.exec(["unzip", wheel_file, "-d", "/user_provided"]).wait()
+
```

Comment:
nothing to do with this line, but maybe you don't have to be in the with block the whole time no?